redis/conn.go

```diff
@@ -127,6 +142,15 @@ func DialNetDial(dial func(network, addr string) (net.Conn, error)) DialOption {
 	}}
 }
 
+// DialNetDialContext specifies a custom dial function with context for creating TCP
+// connections, otherwise a net.Dialer customized via the other options is used.
+// DialNetDialContext overrides DialConnectTimeout and DialKeepAlive.
+func DialNetDialContext(dialContext func(ctx context.Context, network, addr string) (net.Conn, error)) DialOption {
```

Comment:
A more idiomatic name for this would be DialWithContext

Reply:
Do I have to be consistent with DialNetDial?
https://pkg.go.dev/github.com/gomodule/redigo/redis?tab=doc#DialNetDial

Dialer uses DialContext.
https://golang.org/pkg/net/#Dialer.DialContext

There is an example to use WithContext when it takes the context itself as an argument.
https://golang.org/pkg/net/http/#Request.WithContext

Reply:
You are right that `DialNetDialContext` follows the precedent of `DialNetDial` but I think it's quite a mouth full.

What do you think users would prefer?

Reply:
What about `DialContextFunc`?